src/rgw/rgw_lc.cc

```diff
@@ -590,8 +598,42 @@ static int remove_expired_obj(const DoutPrefixProvider *dpp, lc_op_ctx& oc, bool
   ACLOwner bucket_owner;
   bucket_owner.set_id(bucket_info.owner);
 
-  return obj->delete_object(dpp, &oc.rctx, obj_owner, bucket_owner, meta.mtime, false, 0,
-			    version_id, null_yield);
+  if (oc.cct->_conf->rgw_lc_notify) {
```

Comment:
why make it conf based and not based on bucket topics/notifications?

Reply:
if no notifications are are provisioned for the bucket, the cost of this "no-op" would be:
1. in `publish_reserve()` read the bucket topics object, if it does not exist, it would be cached (see: https://github.com/ceph/ceph/pull/35777)
2. in `publish_commit()`, we iterate over the reservations done in (1), if the list is empty we do nothing